src/script/script_instance.cpp

```diff
@@ -569,8 +569,9 @@ bool ScriptInstance::IsPaused()
 
 		case SQSL_STRING: {
 			SlObject(nullptr, _script_byte);
-			static char buf[256];
+			static char buf[std::numeric_limits<decltype(_script_sl_byte)>::max()];
```

Comment:
This feels real dirty. Is `lengthof` not good enough?

Reply:
No, `lengthof` is giving you the numer of elements that are in an array. Here I am not looking for the number of elements in the array, but I am rather looking for the maximum value of `_script_sl_byte` as that's the maximum number of bytes that will be used of the buffer.

Practically `255` or `UINT8_MAX` would suffice, but those will break as soon as someone decides `_script_sl_byte` is to be a UINT16. Ofcourse it could just test whether `_script_sl_byte` is less or equal to `lengthof(buf)`, but that will in its turn cause complaints about dead code since currently there is no way for `_script_sl_byte` to be larger than 255. Only once the type is made larger that becomes a possibility, and only then that code makes any sense.
This way the allocated buffer will always be large enough. Well, for uint16. For uint32 it will probably crash quickly due to allocating too much on the stack. The alternative would be to use `255` or `UINT8_MAX` and then some `static_assert` to ensure the value for `_script_sl_byte` is always within bounds, but that would likely include something similar to what I have written here.